projects/plugins/beta/src/class-plugin.php

```diff
@@ -560,8 +560,11 @@ public function dev_upgrader_response() {
 		$info     = null;
 
 		if ( 'pr' === $dev_info->source && ! isset( $manifest->pr->{$dev_info->id} ) && isset( $manifest->master ) ) {
-			// It's a PR that is gone. Update to master.
+			// It's a PR that is gone. Update to master if master is the main branch.
 			list( , $info ) = $this->get_which_and_info( 'master', '' );
+		} elseif ( 'pr' === $dev_info->source && ! isset( $manifest->pr->{$dev_info->id} ) && isset( $manifest->trunk ) ) {
+			// It's a PR that is gone. Update to trunk if trunk is the main branch.
+			list( , $info ) = $this->get_which_and_info( 'trunk', '' );
```

Comment:
More simplification after doing the "little work":
```suggestion
		if ( 'pr' === $dev_info->source && ! isset( $manifest->pr->{$dev_info->id} ) && isset( $manifest->trunk ) ) {
			// It's a PR that is gone. Update to trunk.
			list( , $info ) = $this->get_which_and_info( 'trunk', '' );
```